src/resources/fileutils.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+#include <dirent.h>
+#include <stdio.h>
+#include <cstring>
+#include <iterator>
+#include "filesystem_littlefs_bbf.h"
+
+#define log(severity, ...) _log_event(severity, log_component_find("Resources"), __VA_ARGS__)
+
+class DIRDeleter {
+public:
+    void operator()(DIR *dir) {
+        closedir(dir);
+    }
+};
+
+class FILEDeleter {
```

Comment:
/nitpick/ may be find a better name ... like `FILEClose` ... when I first saw it in a unique_ptr it looked like we are actually trying to delete the file instead of opening it :wink:

Reply:
I was following codestyle already used by other parts of our project. Try to search for the word `Deleter` and you find a lot of those... deleters.
Also, I don't find it confusing as the template argument is also called deleter. So as soon as I see a class named `SomethingDeleter` I know, it is meant to be used together with std::unique_ptr and a pointer to `Something`

Reply:
ok